cpp/include/cugraph/utilities/thrust_tuple_utils.cuh

```diff
@@ -132,19 +133,44 @@ template <typename... Ts>
 struct is_thrust_tuple<thrust::tuple<Ts...>> : std::true_type {
 };
 
-template <typename TupleType, typename Enable = void>
+template <typename TupleType>
 struct is_thrust_tuple_of_arithmetic : std::false_type {
 };
 
-template <typename TupleType>
-struct is_thrust_tuple_of_arithmetic<TupleType,
-                                     std::enable_if_t<is_thrust_tuple<TupleType>::value>> {
-  static constexpr bool value =
-    detail::is_thrust_tuple_of_arithemetic_impl<TupleType,
```

Comment:
So, if we're not using detail::is_thrust_tuple_of_arithmetic_impl anymore, we should delete unused functions.